src/fs/data/private/spiralratchet.ts

```diff
@@ -43,46 +44,91 @@ async function zero({ webcrypto, crypto }: EncryptionContext): Promise<SpiralRat
 }
 
 
-export async function add1(ratchet: SpiralRatchet, ctx: EncryptionContext): Promise<SpiralRatchet> {
+export async function inc(ratchet: SpiralRatchet, ctx: EncryptionContext): Promise<SpiralRatchet> {
   const { webcrypto } = ctx
 
   if (ratchet.smallCounter >= 255) {
-    // essentially is ratchet-255+256
-    return await add256({
-      ...ratchet,
-      small: await sha(webcrypto, complement(ratchet.medium)),
-      smallCounter: 0,
-    }, ctx)
+    return await next256Epoch(ratchet, ctx)
   }
+
   return Object.freeze({
     ...ratchet,
-    small: await sha(webcrypto, ratchet.small),
+    small: await sha(webcrypto, ratchet.small), // NOTE: this uses the input ratchet
     smallCounter: ratchet.smallCounter + 1,
   })
 }
 
-export async function add256(ratchet: SpiralRatchet, ctx: EncryptionContext): Promise<SpiralRatchet> {
+export async function next256Epoch(ratchet: SpiralRatchet, ctx: EncryptionContext): Promise<SpiralRatchet> {
```

Comment:
This was the big change. `add256` didn't change the smaller values, even though they have to change (based on the _input_ ratchet, not the updated / zeroed out one)

Reply:
Ok, so I think I need to understand this a little better.
I was thinking of the ratchet like an at-least 3-digit number. Let's simplify this and say it's base-10 instead of base-256.
So in base-10, add256 becomes add10.
So, e.g. 123 + 10 = 133.
Why should the least significant digit change when I add 256?
I assume my mental model of the ratchet is wrong. It's not like the ratchet I have in mind wouldn't work as a counter, so I assume it's not (as) secure. Is that what it is, and if so, why?
Trying to learn & understand.

Reply:
Is there a test case that would've failed with the old version? That might illustrate the problem for me very clearly (and maybe make a good test case generally :) )

Reply:
Ok I think I get it now :)

The problem is, we can't have 123 + 10 = 133, because then it'd not be 100% backwards secret! If someone knows 133, then they know 1/3 of the ratchet 10 iterations ago. They'll know it was **3.
Even worse, if we have the second digit do the same thing. So if they know all the second and third digits (i.e. they know *00-*99), then they can go back and forth between any n00 and n99.

We could make a test out of this, I think I'll try that. It'd just check that none of the previous ratchet values has occurred before, not even partially.